### ISSUE TITLE
Some zarr images failed to display correctly

### ISSUE THREAD

**Issue by oeway** [will-moore/omero-imjoy#7]
Hi I noticed that some images are not displayed correctly, for example: https://s3.embassy.ebi.ac.uk/idr/zarr/v0.1/9836846.zarr

**Comment by will-moore** [comment]
Using https://ij.imjoy.io/?plugin=https://gist.github.com/oeway/786f65e8994742ba9cbbd781cc9377cb

Trying to open that image with the NGFF plugin, I'm seeing:

![Screenshot 2021-05-12 at 15 52 53](https://user-images.githubusercontent.com/900055/117996706-490d1000-b33a-11eb-9d8e-b28e52e2d972.png)

because this is a single-Z image and the "Stack to Hyperstak..." command needs a Z-stack.

Is that what you're seeing?
I'm afraid I don't know what the equivalent macro command is if sizeZ is 1. Maybe we don't need that step at all?

**Comment by oeway** [comment]
Right, in that case, we can skip it. I just updated the plugin: https://gist.github.com/oeway/786f65e8994742ba9cbbd781cc9377cb#file-ngff-loader-imjoy-html-L155

However, the image seems wrong:

![image](https://user-images.githubusercontent.com/478667/118543707-e9a56a80-b754-11eb-8da6-afc51cc09087.png)

https://s3.embassy.ebi.ac.uk/idr/zarr/v0.1/9836846.zarr

Any idea?

**Comment by oeway** [comment]
FYI: I have now integrated your code to imagej.js repo: https://github.com/imjoy-team/imagej.js/blob/master/src/zarrUtils.js
If you figured out why that fails, a PR is welcome there ;)

**Comment by oeway** [comment]
Oh, I figured out why this happens, because `getRawChunk` [here](https://github.com/will-moore/omero-imjoy/blob/9ec52590b730d5246201de547ebb09c69147478a/omero_imjoy/templates/omero_imjoy/index.html#L144) only get a chunk of the image plane, when the image plane is small, it can get the entire image, but when it's big, the data chunk won't be sufficient. We will need to load zarr-lite the full version and call `getRaw` instead. For completeness, I made a corrected version here: https://gist.github.com/oeway/786f65e8994742ba9cbbd781cc9377cb

With the latest version you won't need any of these, you can simply do: 
```js
await ij.viewZarr("https://s3.embassy.ebi.ac.uk/idr/zarr/v0.1/9836846.zarr")
```
If you have further changes, e.g. to implement NGFF 0.2, please feel free to change this file here: https://github.com/imjoy-team/imagej.js/blob/master/src/zarrUtils.js  You can also use an ImJoy plugin as playground and later we can integrate into ImageJ.JS.
